File: pet-funeral-system/admin/src/api/ceremony.ts
import { request } from '@/utils/request'

export interface Ceremony {
  id?: number
  orderId?: number
  petId?: number
  petName?: string
  customerName?: string
  ceremonyType?: number
  ceremonyTypeName?: string
  hallNo?: string
  startTime?: string
  endTime?: string
  status?: number
  statusName?: string
  contactName?: string
  contactPhone?: string
  attendeeCount?: number
  specialRequests?: string
  remark?: string
  createdAt?: string
}

export interface CeremonyQuery {
  page?: number
  pageSize?: number
  status?: number
  date?: string
}

// 获取仪式列表
export const getCeremonyList = (params: CeremonyQuery) => {
  return request.get<any>('/ceremonies', { params })
}

// 获取仪式详情
export const getCeremonyDetail = (id: number) => {
  return request.get(`/ceremonies/${id}`)
}

// 创建仪式
export const createCeremony = (data: Ceremony) => {
  return request.post('/ceremonies', data)
}

// 更新仪式
export const updateCeremony = (id: number, data: Ceremony) => {
  return request.put(`/ceremonies/${id}`, data)
}

// 开始仪式
export const startCeremony = (id: number) => {
  return request.post(`/ceremonies/${id}/start`)
}

// 完成仪式
export const completeCeremony = (id: number) => {
  return request.post(`/ceremonies/${id}/complete`)
}

// 取消仪式
export const cancelCeremony = (id: number, reason?: string) => {
  return request.post(`/ceremonies/${id}/cancel`, null, { params: { reason } })
}

// 获取今日仪式
export const getTodayCeremonies = () => {
  return request.get('/ceremonies/today')
}

// 获取某日日程
export const getDaySchedule = (date: string) => {
  return request.get('/ceremonies/schedule/day', { params: { date } })
}

File: pet-funeral-system/backend/src/main/java/com/petfuneral/entity/Ceremony.java
package com.petfuneral.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 告别仪式
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("ceremony")
public class Ceremony extends BaseEntity {

    /**
     * 订单ID
     */
    private Long orderId;

    /**
     * 宠物ID
     */
    private Long petId;

    /**
     * 客户ID
     */
    private Long customerId;

    /**
     * 仪式类型 1-简约告别 2-温馨告别 3-尊享告别
     */
    private Integer ceremonyType;

    /**
     * 仪式厅编号
     */
    private String hallNo;

    /**
     * 开始时间
     */
    private LocalDateTime startTime;

    /**
     * 结束时间
     */
    private LocalDateTime endTime;

    /**
     * 参加人数
     */
    private Integer attendeeCount;

    /**
     * 主持人
     */
    private String host;

    /**
     * 负责员工ID
     */
    private Long staffId;

    /**
     * 特殊要求
     */
    private String specialRequests;

    /**
     * 备注
     */
    private String remark;

    /**
     * 状态 0-待安排 1-已安排 2-进行中 3-已完成 4-已取消
     */
    private Integer status;
}
